packages/notte-sdk/src/notte_sdk/endpoints/page.py

```
@@ -135,31 +171,45 @@ def scrape(
+        # Handle structured data scraping
         structured = response.structured
         if request.requires_schema():
             if structured is None:
-                raise ValueError("Failed to scrape structured data. This should not happen. Please report this issue.")
-            if not structured.success or structured.data is None:
-                return structured
-            if response_format is not None:
-                structured.data = response_format.model_validate(structured.data.model_dump())
+                raise ScrapeFailedError(
+                    "Failed to scrape structured data. This should not happen. Please report this issue."
+                )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)